src/url-pattern.ts

```diff
@@ -128,10 +128,119 @@ function applyInit(o: URLPatternInit, init: URLPatternInit, isPattern: boolean):
   return o;
 }
 
+function escapePatternString(value: string): string {
+  return value.replace(/([+*?:{}()\\])/g, '\\$1');
+}
+
+function escapeRegexpString(value: string): string {
+  return value.replace(/([.+*?^${}()[\]|/\\])/g, '\\$1');
+}
+
+// A utility function to convert a list of path-to-regexp Tokens back into
+// a pattern string.  The resulting pattern should be equivalent to the
+// original parsed pattern, although they may differ due to canonicalization.
+function tokensToPattern(tokens: Token[],
+                         options: TokensToRegexpOptions & ParseOptions): string {
+  let result = "";
+  for (const token of tokens) {
+    // Plain text tokens can be directly added to the pattern string.
+    if (typeof token === 'string') {
+      result += escapePatternString(token);
+      continue;
+    }
+
+    // Tokens without a pattern are also plain text, but were originally
+    // contained in a `{ ... }` group.  There may be a modifier following
+    // the group.  If there is no modifier then we strip the braces off
+    // as they are superfluous.
+    if (token.pattern === '') {
+      if (token.modifier === '') {
+        result += escapePatternString(token.prefix);
+        continue;
+      }
+      result += '{';
```

Comment:
Might be faster to just use template literals here

Reply:
soemthing like result  += `{${escapePatternString(token.prefix)}${token.modifier}`